projects/vdk-core/tests/vdk/internal/builtin_plugins/connection/test_managed_cursor.py

```diff
@@ -163,3 +164,54 @@ def mock_recover(recovery_cursor: RecoveryCursor):
     assert "Could not handle execution exception" == e.value.args[0]
     mock_connection_hook_spec.db_connection_recover_operation.assert_called_once()
     mock_native_cursor.execute.assert_called_once()
+
+
+def test_query_timing_successful_query(caplog):
+    (
+        mock_native_cursor,
+        mock_managed_cursor,
+        _,
+        _,
+        mock_connection_hook_spec,
+    ) = populate_mock_managed_cursor()
+    # set logging level to info
+    mock_managed_cursor._log.level = 20
+    mock_managed_cursor.execute(_query)
+    assert "Successful query duration H:M:S 0:00:" in str(caplog.records)
+
+
+def test_query_timing_recovered_query(caplog):
+    (
+        mock_native_cursor,
+        mock_managed_cursor,
+        _,
+        _,
+        mock_connection_hook_spec,
+    ) = populate_mock_managed_cursor()
+    # set logging level to info
+    mock_managed_cursor._log.level = 20
+    mock_native_cursor.execute.side_effect = [Exception("Mock exception")]
+    mock_managed_cursor.execute(_query)
+    assert "Recovered query duration H:M:S 0:00:" in str(caplog.records)
+
+
+def test_query_timing_failed_query(caplog):
+    (
+        mock_native_cursor,
+        mock_managed_cursor,
+        _,
+        _,
+        mock_connection_hook_spec,
+    ) = populate_mock_managed_cursor()
+    # set logging level to info
+    with mock.patch.object(
+        mock_managed_cursor.__class__, "_recover_operation"
```

Comment:
It's a bad practice to be using any private or protected fields or methods. 
This makes the test more brittle (a small change in teh implementation would break it) 

We want our tests to be as black box as possible (know little of what's going on inside and use only the public interfaces)


There is  `mock_connection_hook_spec.db_connection_recovery_operation` which we can mock around . Wouldn't that work?